dttools/src/catalog_server.c

```diff
@@ -216,19 +223,37 @@ static void make_hash_key(struct jx *j, char *key)
 
 static void handle_updates(struct datagram *update_port)
 {
-	char data[DATAGRAM_PAYLOAD_MAX * 2];
 	char addr[DATAGRAM_ADDRESS_MAX];
 	char key[LINE_MAX];
 	int port;
-	int result;
+	unsigned long data_length, raw_data_length;
 	struct jx *j;
 
 	while(1) {
-		result = datagram_recv(update_port, data, DATAGRAM_PAYLOAD_MAX, addr, &port, 0);
-		if(result <= 0)
+		raw_data_length = datagram_recv(update_port, raw_data, DATAGRAM_PAYLOAD_MAX, addr, &port, 0);
```

Comment:
`datagram_recv` returns -1 if there is no data.

Since `raw_data_length` is unsigned it never terminates the loop.